Formula/otp-cli.rb

```diff
@@ -0,0 +1,26 @@
+class OtpCli < Formula
+  desc "Tool for generate otp code in terminal"
+  homepage "https://github.com/chyroc/otp-cli"
+
+  url "https://github.com/chyroc/otp-cli/releases/download/v0.2.0/otp-cli-0.2.0.tar.gz"
+  sha256 "7e223d86e924e95e722b6c2b69acd0a4963085df2726132fcc6c0731caa93150"
+  head "https://github.com/chyroc/otp-cli"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+
+    bin_path = buildpath/"src/github.com/chyroc/otp-cli"
+    bin_path.install Dir["*"]
+    cd bin_path do
+      system "go", "build", "-o", bin/"otp-cli", "."
+    end
+  end
+
+  test do
+    # "2>&1" redirects standard error to stdout. The "2" at the end means "the
+    # exit code should be 2".
+    assert_match "otp-cli", shell_output("#{bin}/otp-cli -h 2>&1", 2)
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
fix

Reply:
It really isn't, if anything this test is worse because it'll break on the next release.

> We need a test that exercises the some of the functionality of the app.

Reply:
fixed, check version and real usage